misc_interfaces/stubs/colvarproxy_stub.cpp

```diff
@@ -9,6 +9,13 @@
 
 #include <iostream>
 
+#define STUB_OUTPUT_FORCE
```

Comment:
Are there strong reasons not to make this a runtime flag passed to `read_frame_xyz`? It would seem simpler to me, and I don't see a critical performance consideration there.

Reply:
There is already an optional argument for the output prefix. I wonder if I add another one, it may become difficault to maintain in the future due to added complexity, if a 3rd argument parser is not used. What's the best way to add such a runtime flag?

Reply:
There are many choices. Personally I would like to use CLI11 since it is header only and can be used with a single header (https://github.com/CLIUtils/CLI11?tab=readme-ov-file#install). I am not familiar with gflags but it looks like linking to the library is required.

Reply:
Sure, I think that qualifies as "something similar" :-)

A header only library that is only used by our command-line tool should be no trouble at all.

Reply:
> I suggest to give up the two outdated and buggy compilers, and use `-Wshadow` in GCC and Clang. However, this is up to @giacomofiorin .

I checked: we do have this already https://github.com/Colvars/colvars/pull/557  Although it's not quite the same thing as the Sun checks, we are probably quite close to abandoning C++11 as well, which would rule that out.

Dropping all CentOS 7 stuff in an upcoming PR.